ui/settings/static/settings.js

```diff
@@ -0,0 +1,29 @@
+$( document ).ready(function () {
+    let formHasChanged = false;
+    const sectionId = document.getElementById('settings-form').dataset.sectionId;
+
+    $("#settings-form :input").on("change input", function () {
+        formHasChanged = true;
+    });
+
+    $("#cancel, #settings-icon").click(function (e) {
+        if(formHasChanged) {
+            e.preventDefault();
+            $("#unsavedChangesModal").modal("show");
+        }
+    });
+
+    $("#customFontInput").click(function () {
+        const radio = document.getElementById('customFontRadio');
+        const input = document.getElementById('customFontInput');
+        radio.checked = true;
+    });
+
```

Comment:
Das `document.getElementById` kann in dieser Datei an ein paar Stellen verkürzt werden.

```suggestion
    $("#customFontInput").click(function () {
        $("#customFontRadio").prop("checked", true);
    });
    
```